include/fiction/algorithms/physical_design/graph_oriented_layout_design.hpp

```diff
@@ -0,0 +1,1523 @@
+                        }
+                    }
+                    for (const auto& [next, cost] : expansion.first)
+                    {
+                        if (ssg.cost_so_far.find(next) == ssg.cost_so_far.cend() || cost < ssg.cost_so_far[next])
+                        {
+                            ssg.cost_so_far[next] = cost;
+                            double priority       = cost;
+                            ssg.frontier.put(next, priority);
+                        }
+                    }
+                }
+            }
+            for (auto& ssg : ssg_vec)
+            {
+                if (ssg.frontier_flag)
+                {
+                    if (!ssg.frontier.empty())
+                    {
+                        ssg.current_vertex = ssg.frontier.get();
+                    }
+                    else
+                    {
+                        ssg.frontier_flag = false;
+                    }
+                }
+            }
+
+            timeout_limit_reached =
+                std::none_of(ssg_vec.cbegin(), ssg_vec.cend(), [](const auto& ssg) { return ssg.frontier_flag; });
+
+            const auto end = std::chrono::high_resolution_clock::now();
+            const auto duration_ms =
+                static_cast<uint64_t>(std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count());
+
+            if (duration_ms >= timeout)
+            {
+                timeout_limit_reached = true;
+            }
+        }
+
+        return best_lyt;
+    }
+
+  private:
+    /**
+     * Alias for an obstruction layout based on the given layout type.
+     */
+    using ObstrLyt = obstruction_layout<Lyt>;
+    /**
+     * The network to be placed and routed.
+     */
+    tec_nt ntk;
+    /**
+     * Parameters.
+     */
+    graph_oriented_layout_design_params ps;
+    /**
+     * Statistics.
+     */
+    graph_oriented_layout_design_stats& pst;
+    /**
+     * Timeout limit (in ms).
+     */
+    uint64_t timeout;
+    /**
+     * Start time.
+     */
+    std::chrono::time_point<std::chrono::high_resolution_clock> start;
+    /**
+     * Number of search space graphs.
+     */
+    uint64_t num_search_space_graphs;
+    /**
+     * Vector of search space graphs.
+     */
+    std::vector<search_space_graph<ObstrLyt>> ssg_vec;
+    /**
+     * Count evaluated paths in the search space graphs.
+     */
+    uint64_t num_evaluated_paths{0ul};
+    /**
+     * Keep track of the maximum number of placed nodes.
+     */
+    uint64_t max_placed_nodes{0ul};
+    /**
+     * Current best solution w.r.t. area.
+     */
+    uint64_t best_solution{std::numeric_limits<uint64_t>::max()};
+    /**
+     * Current best solution w.r.t. area after relocating POs.
+     */
+    uint64_t best_optimized_solution{100000};
+    /**
+     * Flag indicating if an initial solution was already found, and that other search space graphs should be pruned.
+     */
+    bool improv_mode{false};
```

Comment:
I'm not certain what `improv_mode` stands for. Can we rename this variable to something more telling?